ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -199,6 +203,68 @@ def batched_step_result_from_proto(
     return BatchedStepResult(obs_list, rewards, done, max_step, agent_id, action_mask)
 
 
+def proto_from_batched_step_result(
+    batched_step_result: BatchedStepResult
+) -> List[AgentInfoProto]:
+    agent_info_protos: List[AgentInfoProto] = []
+    for agent_id in batched_step_result.agent_id:
+        agent_id_index = batched_step_result.get_index(agent_id)
+        reward = batched_step_result.reward[agent_id_index]
+        done = batched_step_result.done[agent_id_index]
+        max_step_reached = batched_step_result.max_step[agent_id_index]
+        agent_mask = None
+        if batched_step_result.action_mask is not None:
+            mask = batched_step_result.action_mask[0]
```

Comment:
I could not get this to work otherwise.  The protobuf was not expecting a list of action masks as in the StepResult.  Thoughts @ervteng ?